rfcs/convergence/props-consistency-check.md

```diff
@@ -0,0 +1,391 @@
+```
+
+## Size properties
+
+> Some components use different sets of semantic sizes. Avatar uses a set of numbers.
+> The property name for filling available space is sometimes block and sometimes fluid. The fit property value of cover could also be considered filling available space.
+
+AccordionHeaderCommons
+
+```ts
+// AccordianHeaderSize = 'small' | 'medium' | 'large' | 'extra-large'
+size?: AccordionHeaderSize;
+```
+
+AvatarProps
+
+```ts
+// AvatarSizeValue = 20 | 24 | 28 | 32 | 36 | 40 | 48 | 56 | 64 | 72 | 96 | 120 | 128
+size: AvatarSizeValue;
+```
+
+BadgeProps
+
+```ts
+// BadgeSize = 'smallest' | 'smaller' | 'small' | 'medium' | 'large' | 'larger' | 'largest'
+
+size: BadgeSize;
+showZero: boolean;
+```
+
+CounterBadgeProps
+
+```ts
+showZero: boolean;
+```
+
+ButtonProps
+
+```ts
+block: boolean;
+size: 'small' | 'medium' | 'large';
+```
+
+ImageProps
+
+```ts
+fit: 'none' | 'center' | 'contain' | 'cover';
+fluid: boolean;
+```
+
+LabelProps
+
+```ts
+size: 'small' | 'medium' | 'large';
+```
+
+PopoverProps
+
+```ts
+//PopoverSize = 'small' | 'medium' | 'large'
+size: PopoverSize;
+```
+
+## Appearance properties
+
+> Some components have mutually exclusive booleans and others use discriminated unions.
+> Some components define a color property that are a set of colors, others are a set semantic names.
+> Some naming is inconsistent (e.g. noArrow, pointing)
+
+AvatarProps
+
+```ts
+activeDisplay: 'ring' | 'shadow' | 'glow' | 'ring-shadow' | 'ring-glow';
+// AvatarNamedColor = 'darkRed' | 'cranberry' ...
+color: 'neutral' | 'brand' | 'colorful' | AvatarNamedColor;
+square: boolean;
+```
+
+BadgeProps
+
+```ts
+// BadgeAppearance = 'filled' | 'outline' | 'ghost' | 'tint'
+appearance: BadgeAppearance;
+// BadgeShape = 'rounded' | 'square' | 'circular'
+shape: BadgeShape;
+// BadgeColors =  | 'brand'  | 'danger'  | 'severe'  | 'warning'  | 'success'  | important'  | 'informative'  | 'subtle'
+color: BadgeColors;
+```
+
+CounterBadgeProps
+
+```ts
+dot: boolean;
+```
+
+ButtonProps
+
+```ts
+circular: boolean;
+outline: boolean;
+primary: boolean;
+subtle: boolean;
+transparent: boolean;
+```
+
+DividerProps
+
+```ts
+appearance: 'default' | 'subtle' | 'brand' | 'strong';
+important: boolean;
+inset: boolean;
+```
+
+ImageProps
+
+```ts
+bordered: boolean;
+circular: boolean;
+rounded: boolean;
+```
+
+LabelProps
+
+```ts
+strong: boolean;
+```
+
+LinkProps
+
+```ts
+secondary: boolean;
+```
+
+PopoverProps
+
+```ts
+brand: boolean;
+inverted: boolean;
+noArrow: boolean;
+```
+
+TooltipProps
+
+```ts
+inverted: boolean;
+pointing: boolean;
+```
+
+## Events and component state
+
+> The PopoverProps onChangeData event has PopoverState as the type of the data. It maybe difficult to maintain consistency between expected event payload and the state used internally to the component to render. The state also includes refs to elements that may constitute and implementation detail and create confusion with slots.
```

Comment:
The actual type is `export interface OnOpenChangeData extends Pick<PopoverState, 'open'> {}` This does not extend all of `PopoverState` but just picks the `open` property from that state

Reply:
Ah - missed the Pick<> there.  We should probably still define event args type though rather than overexpose the state type.